### PR TITLE
feat(ruby): support to call method using symbol as well in Message_index

### PR DESCRIPTION
**Pull request by bajajneha27**
b/229817649

### CODE REVIEW

File: ruby/ext/google/protobuf_c/message.c [hunk 1]
```diff
@@ -910,14 +910,22 @@ static VALUE Message_freeze(VALUE _self) {
...
 
-  Check_Type(field_name, T_STRING);
-  field = upb_MessageDef_FindFieldByName(self->msgdef, RSTRING_PTR(field_name));
+  if (TYPE(field_name) == T_STRING) {
+    name = RSTRING_PTR(field_name);
+  } else if (TYPE(field_name) == T_SYMBOL) {
+    name = RSTRING_PTR(rb_id2str(SYM2ID(field_name)));
+  } else {
+    rb_raise(rb_eArgError,
+             "Expected string or symbols as hash keys when accessing a field's "
```
Comment (haberman) [inline ruby/ext/google/protobuf_c/message.c:925]: Let's not use the word "hash" here. It implies that there is a Ruby Hash object underneath, which there is not.

I think the error message words if you just remove the word "hash."

**Review by haberman — CHANGES_REQUESTED** [review]
Please add a test.

**Comment by bajajneha27** [comment]
> Please add a test.

I added a test. I'm not sure about the reasons of failure. Can someone from your team help me out here?

**Comment by JasonLunn** [comment]
@bajajneha27 - can you merge `main` into your branch? I went to go look at the failing logs but they're no longer available.

**Comment by bajajneha27** [comment]
> @bajajneha27 - can you merge `main` into your branch? I went to go look at the failing logs but they're no longer available.

Done. Thanks for looking  into it.

**Comment by JasonLunn** [comment]
Ruby tests are failing with a segfault in `ruby/tests/common_tests.rb` line 901 and `ruby/compatibility_tests/v3.0.0/tests/basic.rb` line 926. Do these reproduce for you locally?

**Comment by bajajneha27** [comment]
> ruby/compatibility_tests/v3.0.0/tests/basic.rb

I see `common_test` one in my local :
```
/usr/local/google/home/bajajnehaa/gcp/protobuf/ruby/tests/common_tests.rb:901:in `test_bad_field_names'
/usr/local/google/home/bajajnehaa/gcp/protobuf/ruby/tests/common_tests.rb:901:in `[]'
```

But I don't see `basic.rb` in the error log.
I ran `rake test` to run the tests

**Pull request closed without merging**